hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OutboundResponseHandler.java

```diff
@@ -87,15 +119,123 @@ public boolean send(Response response, Address target) {
 
         byte[] bytes = serializationService.toBytes(response);
         Packet packet = new Packet(bytes, -1)
-                .setPacketType(Packet.Type.OPERATION)
+                .setPacketType(OPERATION)
                 .raiseFlags(FLAG_OP_RESPONSE);
 
         if (response.isUrgent()) {
             packet.raiseFlags(FLAG_URGENT);
         }
 
+        return transmit(target, packet);
+    }
+
+    private boolean sendNormalResponse(Address target, long callId, int backupAcks, boolean urgent, Object value) {
+        checkTarget(target);
+
+        Packet packet = toNormalResponsePacket(callId, (byte) backupAcks, urgent, value);
+
+        return transmit(target, packet);
+    }
+
+    Packet toNormalResponsePacket(long callId, int backupAcks, boolean urgent, Object value) {
+        byte[] bytes;
```

Comment:
They can still be called; I'm using them even to verify if the 'optimized' serialization does the same thing as serialization NormalResponse. 

Eventually I would like to drop all the responses since they are not truly needed; but I tried this a few time and it leads to a lot of code change.

```
I like the optimisation, I'm just worrying that someone forgets that we duplicate the logic from the DataSerializableSerializer and EnterpriseDataSerializableSerializer here.
```
They can't be changed.. at least not in the 3.x timeline due to minor version compatibility.

```
Another idea - we move this into a new method on the internal serialization service?
```
I don't think we should get a dependency of something as specific as response into something generic as serialization. It would cause a cycle (tight coupling).